src/modules/infobox/components/styling/FeatureStyleEdit.vue

```diff
@@ -28,22 +29,51 @@ const props = defineProps({
 })
 const { feature, readOnly } = toRefs(props)
 
-const description = computed({
-    get() {
-        return feature.value.description
-    },
-    set(value) {
-        store.dispatch('changeFeatureDescription', { feature: feature.value, description: value })
-    },
+const title = ref(feature.value.title)
+const description = ref(feature.value.description)
+
+// Update the UI when the feature changes
+watch(
+    () => feature.value.title,
+    (newTitle) => {
+        title.value = newTitle
+    }
+)
+
+watch(
+    () => feature.value.description,
+    (newDescription) => {
+        description.value = newDescription
+    }
+)
+
+// The idea is watching the title and the description.
+// Put a debounce on the update of the feature so that we can compare with the current UI state
+// If the value is the same as in the UI, we can update the feature
+watch(title, (newTitle) => {
+    const debounceTitleUpdate = debounce(updateFeatureTitle, 300)
+    debounceTitleUpdate(newTitle)
 })
-const text = computed({
-    get() {
-        return feature.value.title
-    },
-    set(value) {
-        store.dispatch('changeFeatureTitle', { feature: feature.value, title: value })
-    },
+watch(description, (newDescription) => {
+    const debounceDescriptionUpdate = debounce(updateFeatureDecription, 300)
+    debounceDescriptionUpdate(newDescription)
 })
+
+function updateFeatureTitle(previousTitle) {
+    if (previousTitle === title.value) {
```

Comment:
If the debounce method is working fine this `if` should not be needed, I had once an issue with the debounce method where its closure was somehow not working and therefore not debouncing and I couldn't find out why the closure was not working and I ended up using my own debounce implementation instead of the debounce method, maybe you have the same issue here ?

Reply:
I could quickly checked that in your case the debounce method closure does not work like it should work in App.vue. I don't know why ? Would be great if you find out why otherwise you would need to create your own debounce implementation.

Reply:
But in the end, we still need to check if the text in the input field are not changing, right? That's why we will still need the `if` statement. If we don't check it, then the dispatch will be called as many as the text change, just with delay.

Reply:
No the idea of the debounce method is that if it is called again before the delay the previous call is dropped, see https://github.com/geoadmin/web-mapviewer/blob/develop/src/utils/debounce.js
Somehow if you define the debounce method inside a watcher, the javascript closure of the debounce method don't work and it doesn't debounce. In this case the debounce is not fully working and is just delaying the call, therefore you need the `if`

Reply:
Somehow my commit seems to break a cypress test on 3d, @ismailsunni could you try to rebase on top of develop and see if it fails again ?

Reply:
Sure!
And yes, after I read it again, you are right, I shouldn't need the `if`. Thanks for the explanation.